src/3rdparty/weakjack/CMakeLists.txt

```diff
@@ -7,6 +7,6 @@ IF(LMMS_HAVE_WEAKJACK)
 	
 	# Library stub for AppImages running on systems without jack
 	ADD_LIBRARY(weakjack MODULE weakjack/weak_libjack.c weakjack/weak_libjack.h)
-	TARGET_INCLUDE_DIRECTORIES(weakjack PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/weakjack ${JACK_INCLUDE_DIRS})
+	TARGET_INCLUDE_DIRECTORIES(weakjack PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/weakjack ${CMAKE_CURRENT_SOURCE_DIR}/../jack2/common ${JACK_INCLUDE_DIRS})
```

Comment:
Could you explain why did you add this?

Reply:
because using weakjack the real jack is not needed, only the header files.

but in that case the right submodule is actually https://github.com/jackaudio/headers

Reply:
For Ardour windows builds, we use http://ardour.org/files/deps/jack_win3264.tar.xz

It's a re-zip of jack headers, .def, .lib, and  32/64bit .dll  from jackd 1.9.10 with a pkg-config file added for convenience,
The files were made using sletz' last release of jack for windows back in 2012:
* https://dl.dropboxusercontent.com/u/28869550/Jack_v1.9.10_32_setup.exe
* https://dl.dropboxusercontent.com/u/28869550/Jack_v1.9.10_64_setup.exe

With weakjack one only needs the headers, but the other files come in handy for unit-tests.

Reply:
> Could you explain why did you add this?

This part seems answered.

> the right submodule is actually https://github.com/jackaudio/headers

Done via b852863, but for the record, now the submodule folder is mismatched from the project name and this is inconsistent with the [other submodules we clone](https://github.com/LMMS/lmms/issues/3963) due to needing a `jack/` prefix which `jack2` was already providing but I digress. :)

> With `weakjack` one only needs the headers, but the other files come in handy for unit-tests.

We're moving most of our Windows system to use `vcpkg` bootstrapping and building on-the-fly.  I don't love this strategy from a time-to-build-perspective, but it certainly avoids the pre-built fetching on Windows, which in 2020 still seems to lack any consistent dependency system.  For the scope of this PR this is all moot since we don't have JACK unit tests (yet) but I'm sure this will come back up in the future.  Anyway, thanks kindly for sharing.  I had initially planned on making a [viable dependency system using precompiled binaries and scoop.sh](https://github.com/lukesampson/scoop-extras/pulls?q=is%3Apr+author%3Atresf+is%3Aclosed), but never came to fruition.  For now, `vcpkg` seems to be the least cluster**** solution.  I mention all of this because as a project, we're always interested in how others do it.  LMMS has had a rocky history of building on Windows and we greatly appreciate the info.

Reply:
@falkTX the recommended header repo throws the following error in AppVeyor:

```
C:\projects\lmms\src\3rdparty\jackaudio\jack/types.h(25): fatal error C1083: Cannot open include file: 'pthread.h':
```

Which seems to be referenced here:
https://github.com/jackaudio/headers/blob/master/types.h

... but not here:
https://github.com/jackaudio/jack2/blob/develop/common/jack/types.h

Is this just a matter of the headers repo needing updating?

Reply:
> > Could you explain why did you add this?

> This part seems answered.

As usually @PhysSong's right 🤦 , I had forgotten to delete the redundant include, the variable is already populated earlier.  Patching now...

Reply:
Done via 9c1cf61.  Moving other conversation points to another line-number for context.

Reply:
> [](https://github.com/jackaudio/jack2/blob/develop/common/jack/types.h)
> 
> Is this just a matter of the headers repo needing updating?

yeah, the headers at the moment target jack1.

Reply:
> yeah, the headers at the moment target jack1.

Should I revert then?

Reply:
yes sorry.